--- a/acinclude.m4
++ b/acinclude.m4
@@ -1495,7 +1495,7 @@ AC_DEFUN([GMP_PROG_CXX_WORKS_PART],
 int main (void) { return 0; }
 EOF
   echo "Test compile: [$2]" >&AC_FD_CC
-  gmp_cxxcompile="$1 conftest.cc >&AC_FD_CC"
+  gmp_cxxcompile="$1 conftest.cc $LDFLAGS >&AC_FD_CC"
   if AC_TRY_EVAL(gmp_cxxcompile); then
     if test "$cross_compiling" = no; then
       if AC_TRY_COMMAND([./a.out || ./b.out || ./a.exe || ./a_out.exe || ./conftest]); then :;
